src/packages/record-screen/index.ts

```diff
@@ -0,0 +1,124 @@
+import writeToStore from '@/packages/write-to-store';
+import Features from '@/packages/features';
+import FeatureToggleStore from '@/packages/feature-toggle-store';
+
+export interface ScreenRecorderOptions {
+  videoName?: string;
+  encodingType?: string,
+}
+
+export class ScreenRecorder<T extends ScreenRecorderOptions> {
+  private recorder;
+  private data;
+  private options: T;
+
+  constructor(options: T) {
+    this.options = options;
+  }
+
+  public async start(): Promise<void> {
+    if (!this.isEnabled() || !this.isBrowserCompatible()) {
+      return;
+    }
+
+    const constraints = { video: { mediaSource: "screen" } };
+
+    const stream = await this.getDisplayMedia(constraints);
+
+    this.recorder = new window['MediaRecorder'](stream);
+    this.data = [];
+
+    this.recorder.ondataavailable = (evt) => this.data.push(evt.data);
+
+    this.recorder.start();
+
+    await new Promise((resolve, reject) => {
+      this.recorder.onstop = () => {
+        this.recorderOnstop();
+        resolve();
+      };
+      this.recorder.onerror = reject;
+    });
+  }
+
+  public stop() {
+    if (!this.recorder) {
+      return;
+    }
+
+    this.recorder.stop();
+  }
+
+  private recorderOnstop() {
+    // create file to DL
+    const node = document.createElement('a');
+    node.setAttribute('id', 'video');
+    node.innerText = this.options.videoName;
+
+    const video = new Blob(this.data, { type: this.options.encodingType });
```

Comment:
we should export this to the `PerryReportInfo` somehow, so that we can leave this as a plugin work to actually encode it as it prefers